web-console/developer/pipelines/pipelines.md

```diff
@@ -1,5 +1,314 @@
+Overview
+Logs
+Pipeline Run Actions
+Delete Pipeline Run
+Status Labels, Icons & Colors
+Task Status Component
+Examples
+Behavior
+Status Colors and Icons
+Pipeline Visualization
+Specs
+Pipeline Run Visualization
+Specs
+
+
+Overview
+The Pipelines navigation item is only shown in the developer perspective.  
+When no projects exist, clicking Pipelines will show an empty state:
+When a project is not in context, clicking Pipelines will prompt the user to select from a list of projects:
+
+Pipelines Resources
+Navigation item: Pipelines  
+Secondary masthead is visible and includes the project selector 
+Page title is Pipelines
+Actions
+There are no page level actions
+There are no pipeline actions, thus no kebabs are in the list view
+Filters
+OpenShift filter toggle component is present
+OpenShift filter by name component is present
+Columns
+
+Column Name
+Details
+Sortable
+Filterable
+Name
+Resource badge + pipeline name are presented as a link.  Clicking on the link navigates to the Pipeline Details View
+Yes
+No
+Last Pipeline Run
+Resource badge + name of last pipeline run
+
+Clicking on the link navigates to the Pipeline Run Details View
+Yes
+No
+Last Run Status
+Running (use fa-refresh)
+Succeeded (use pficon-ok)
+Failed (use pficon-error-circle-o)
+Pending (fa-hourglass-half)
+Cancelled (fa-ban)
+
+Follow these guidelines when displaying status
+Yes
+Yes
+Last Task Status
+Follow these guidelines when displaying the task status
+
+
+
+
+Last Run Started
+Follow these guidelines when displaying time
+Yes
+No
+
+
+When a project is in context, clicking “Pipelines” will display the Pipelines Resources in that project.
+
+## Pipelines Resource Details
+### Pipelines - Overview tab
+![Pipelines Overview tab](img/PL-Details-Overview.png)
+
+### Pipelines - YAML tab
+![Pipelines YAML tab](img/PL-Details-YAML.png)
+
+### Pipelines - Pipeline Runs tab
+![Pipelines PLR tab](img/PL-Details-PLR.png)
+
+### Pipelines - Parameters tab
+![Pipelines Parameters tab](img/PL-Details-Parameters.png)
+
+### Pipelines - Resources tab
+![Pipelines Resources tab](img/PL-Details-Resources.png)
+
+
+There are 2 columns in the Pipelines resource tab:
+Name - resource name
+Resource Type - a dropdown including git, image, cluster & storage
+
+
+Adding a Resource
+The general behavior of adding a resource is consistent with the Environments tab of many workloads pages.  
+
+When the user clicks Add Resource
+a new row is added
+hints are shown in the input fields
+An info inline notification is shown 
+
+
+
+
+
+
+When the user clicks Save
+An success inline notification is shown 
+
+
+
+
+Removing a Resource
+When the user hovers over the Remove Resource icon
+A Remove Resource tooltip should be displayed
+![Remove a resource-00](img/PL-Details-Resources-RemoveResource.png)
+
+When the user clicks the Remove Resource icon
+The row is removed
+An info inline notification is shown 
+![Remove a resource-01](img/PL-Details-Resources-RemoveResource01.png)
```

Comment:
Is this a new pattern? We have some similar removal use cases and are using a confirmation modal and saving each action on confirm rather than an overall Save/Reload option. Maybe a question for @beanh66 too :)

Reply:
This is the pattern being used in the Environment tab of Detail pages.  We just duplicated it :) . We also asked Stacey for text and improved the readability of that reload test.  We should consider it in the design conventions for sure!

Reply:
Thanks @serenamarie125!! I hadn't seen that one yet. Totally agreed on considering a convention here.